### PR TITLE
Fix API Doc generation

### PR DESCRIPTION
**Pull request by sc-addypathania**
The typescript version present in the root package.json file was incompatible with the types within the packages which caused the api generation to fail.

- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated

## Testing Details

- [ ] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ Our versioning strategy is as follows:
 * `[templates/nextjs-multisite]` Fix skipped site info fetch ([#1434](https://github.com/Sitecore/jss/pull/1434))
 * `[angular]` Fix app build errors. Webpack version is locked at 5.78 due to https://github.com/webpack/webpack/issues/16981 ([#1448](https://github.com/Sitecore/jss/pull/1448))
 * `[sitecore-jss-nextjs]` Fix middleware redirect when the target use regexp with querystring ([#1449](https://github.com/Sitecore/jss/pull/1449))
+* `Fix API Doc generation ([#1464](https://github.com/Sitecore/jss/pull/1464))
```
Comment (illiakovalenko) [inline CHANGELOG.md:49]: I think it's better moving that under "Chores" section, not really a bugfix

File: package.json [hunk 2]
```diff
@@ -30,9 +30,9 @@
     "eslint-plugin-prettier": "^3.3.0",
     "lerna": "^5.6.2",
     "prettier": "^1.14.3",
-    "typedoc": "^0.22.5",
+    "typedoc": "^0.23.28",
     "typedoc-plugin-markdown": "^3.11.3",
-    "typescript": "~4.3.5"
+    "typescript": "~4.7.4"
```
Comment (illiakovalenko) [inline package.json:35]: Few points here:
* _typescript_ version should be the same like we are using in base packages (~4.9.3)
* latest _typedoc_ version is 0.24.6, should we just use latest?

**Review by illiakovalenko — CHANGES_REQUESTED** [review]
Extra question regarding warnings during typedoc generation
is it fixed? I guess it's also expected to be done

File: package.json (hunk 2 above)
Comment (sc-addypathania) [reply package.json:35]: I tried updating them to the latest version but then `sitecore-jss-angular-schematics` package failed for doc generation.
I tried updating typescript in `sitecore-jss-angular-schematics` but that lead to another error. This was the only stable combination until now and we are using typescript 4.7 in most of the packages.

**Comment by sc-addypathania** [comment]
> Extra question regarding warnings during typedoc generation is it fixed? I guess it's also expected to be done

When we generate docs from the root there are no warnings. Docs generated for individual packages has warnings which is the case for all the packages. Should this be covered in a separate PBI ?

File: package.json (hunk 2 above)
Comment (illiakovalenko) [reply package.json:35]: As a workaround it's fine for now

**Pull request merged**